### DISCUSSION TITLE
ArgTypes configuration for array of objects

### DISCUSSION THREAD

**Discussion by alexandermirzoyan** [storybookjs/storybook#18680] [General]
Hi all,
I have a web component which accepts an array of objects with a custom scheme. Is there any way how can I configure the pattern.
I want that icon property would accept only a select control.

Current config that I use.
```
argTypes: {
    options: {
      control: { type: 'array' },
      title: { control: 'text' },
      icon: { control: 'select', options: ICONS },
      action: { control: 'text' },
      link: { control: 'text' },
    },
  },
```
![image](https://user-images.githubusercontent.com/48055294/178282666-1caba584-e9f7-495d-9cfd-ab8b87d2d3b5.png)

**Comment by blowsie** [comment]
```
argTypes: {
      control: { type: 'array' },
      title: { control: 'text' },
      icon: { control: 'select', options: ICONS },
      action: { control: 'text' },
      link: { control: 'text' },
  },

**Reply by alexandermirzoyan** [reply]
Thank you. Now I stopped working on that project, but if anyone other tried and it helped please let us know.
